Network/DNS/Resolver.hs

```diff
@@ -342,15 +369,37 @@ lookupRawInternal rcv ad rlv dom typ = do
               Nothing  -> loop query checkSeqno (cnt + 1) False
               Just res -> do
                   let valid = checkSeqno res
-                  if valid then
-                      return $ Right res
-                    else
-                      loop query checkSeqno (cnt + 1) False
+                  case valid of
+                      False  -> loop query checkSeqno (cnt + 1) False
+                      True | False <- trunCation $ flags $ header res
```

Comment:
I did not know PatternGuards is now standard. Actually, I checked the language extensions in this file but did not found PatternGuards, so I could not understand this code.
Anyway, thank you for your explanation. I will release a new version AS IS from now on.